app/src/main/java/com/larryhsiao/nyx/view/internals/BrowseFragment.java

```diff
@@ -0,0 +1,157 @@
+package com.larryhsiao.nyx.view.internals;
+
+import android.content.Context;
+import android.content.Intent;
+import android.net.Uri;
+import android.os.Bundle;
+import android.view.KeyEvent;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.webkit.MimeTypeMap;
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.appcompat.app.AlertDialog;
+import androidx.core.content.ContextCompat;
+import androidx.core.content.FileProvider;
+import androidx.fragment.app.Fragment;
+import androidx.recyclerview.widget.LinearLayoutManager;
+import androidx.recyclerview.widget.RecyclerView;
+import com.larryhsiao.nyx.R;
+import com.larryhsiao.nyx.view.diary.attachment.ViewAttachmentIntent;
+import com.silverhetch.aura.AuraFragment;
+import com.silverhetch.clotho.utility.comparator.FileComparator;
+import com.silverhetch.clotho.utility.comparator.StringComparator;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.List;
+
+import static android.Manifest.permission.READ_EXTERNAL_STORAGE;
+import static android.Manifest.permission.WRITE_EXTERNAL_STORAGE;
+import static android.view.KeyEvent.KEYCODE_BACK;
+import static java.util.Objects.requireNonNull;
+
+/**
+ * Fragment that browse the file system.
+ */
+public class BrowseFragment extends AuraFragment {
+    private final static String KEY_PATH = "KEY_PATH";
+
+    public static Fragment newInstance(final String path) {
+        final BrowseFragment frag = new BrowseFragment();
+        final Bundle args = new Bundle();
+        args.putString(KEY_PATH, path);
+        frag.setArguments(args);
+        return frag;
+    }
+
+    private File root = null;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Fields should be declared at the top of the class, before any method declarations, constructors, initializers or inner classes.](https://app.codacy.com/app/LarryHsiao/Nyx/pullRequest?prid=4677927)